src/lib.rs

```diff
@@ -1,11 +1,14 @@
 use crate::error::{Result, TokenGenerationError};
```

Comment:
i think some top level crate documentation (`//! ...` would be nice, especially with some examples